src/main/java/com/artipie/rpm/meta/XmlMetaJoin.java

```diff
@@ -0,0 +1,129 @@
+import javax.xml.stream.XMLEventFactory;
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLEventWriter;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.XMLEvent;
+
+/**
+ * Joins two meta xml-files.
+ * @since 0.9
+ */
+public final class XmlMetaJoin {
+
+    /**
+     * Tag.
+     */
+    private final String tag;
+
+    /**
+     * Ctor.
+     * @param tag Metatag
+     */
+    public XmlMetaJoin(final String tag) {
+        this.tag = tag;
+    }
+
+    /**
+     * Appends data from part to target.
+     * @param target Target
+     * @param part File to append
+     * @throws IOException On error
+     */
+    public void merge(final Path target, final Path part) throws IOException {
+        final Path res = target.getParent().resolve(
+            String.format("%s.joined", target.getFileName().toString())
+        );
+        try (OutputStream out = Files.newOutputStream(res)) {
+            final XMLEventWriter writer = XMLOutputFactory.newInstance().createXMLEventWriter(out);
+            this.writeFirstPart(target, writer);
+            this.writeSecondPart(part, writer);
+            writer.close();
+        } catch (final XMLStreamException ex) {
+            throw new IOException(ex);
+        }
+        Files.move(res, target, StandardCopyOption.REPLACE_EXISTING);
+    }
+
+    /**
+     * Writes the first part.
+     * @param target What to write
+     * @param writer Where to write
+     * @throws IOException On error
+     * @throws XMLStreamException On error
+     */
+    private void writeFirstPart(final Path target, final XMLEventWriter writer)
+        throws IOException, XMLStreamException {
+        try (InputStream in = Files.newInputStream(target)) {
+            final XMLEventReader reader = XMLInputFactory.newInstance().createXMLEventReader(in);
+            writer.add(reader.nextEvent());
+            writer.add(XMLEventFactory.newFactory().createSpace("\n"));
+            while (reader.hasNext()) {
+                final XMLEvent event = reader.nextEvent();
+                if (!(event.isEndElement()
+                    && event.asEndElement().getName().getLocalPart().equals(this.tag))
+                    && !event.isEndDocument()
+                ) {
+                    writer.add(event);
+                }
+            }
+            reader.close();
+        }
+    }
+
+    /**
+     * Writes second part.
+     * @param part What to write
+     * @param writer Where to write
+     * @throws IOException On error
+     * @throws XMLStreamException On error
+     */
+    private void writeSecondPart(final Path part, final XMLEventWriter writer)
+        throws IOException, XMLStreamException {
+        try (InputStream in = Files.newInputStream(part)) {
+            final XMLEventReader reader = XMLInputFactory.newInstance().createXMLEventReader(in);
```

Comment:
@olenagerasimova I'd put it to `try ()` too (if it's closeable)

Reply:
@g4s8 no, unfortunately not((